ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/HiveAggregateInsertDeleteIncrementalRewritingRule.java

```diff
@@ -139,7 +139,14 @@ protected IncrementalComputePlanWithDeletedRows createJoinRightInput(RelOptRuleC
       switch (aggregateCall.getAggregation().getKind()) {
         case COUNT:
           aggFunction = SqlStdOperatorTable.SUM;
-          argument = relBuilder.literal(1);
+
+          // count(*)
+          if (aggregateCall.getArgList().isEmpty()) {
+            argument = relBuilder.literal(1);
+          } else {
+            // count(<column_name>)
+            argument = genArgumentForCountColumn(relBuilder, rexBuilder, aggInput, aggregateCall);
```

Comment:
notes: I think you could access `rexBuilder` from `relBuilder`

I think you could also push the encolsing `if` into this function; or inline the whole function - but right now you have some "count argument" related logic here and there too